csv-generator/csv_generator/reviewer_sharing_name.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import csv
+import logging
+import os
+from itertools import islice
+import re
+from typing import List
+
+from tqdm import tqdm
+
+from .consumers.utils import timestamp_to_epoch, clean_msid
+
+LOGGER = logging.getLogger(__name__)
+
+CSV_TEMP_FILE = 'temp.csv'
+CSV_OUTPUT_FILE = 'reviewer_identity_revealed.csv'
+DEFAULT_OUTPUT_DIR = '../output'
+HEADERS = [
+    'create_date',
+    'source_file_name',
+    'msid',
+    'version_position_in_ms',
+    'reviewer_person_id',
+    'share_name'
+]
+ROWS_TO_SKIP = 4
+
+
+def extract_create_date(input_data: str) -> int:
+    """Extracts create date from file name and
+    returns its epoch value.
+
+    example:
+
+    >>> extract_create_date('...identity_revealled_2018_07_13_eLife.csv')
+    1531440000
+
+    :param input_data: str
+    :return: int
+    """
+    date_str = ''
+    result = re.search(r'(\d{4}_\d{2}_\d{2})', input_data)
+    if result:
+        date_str = result.group()
+
+    return timestamp_to_epoch(date_str)
+
+
+def extract_version_position(input_data: str) -> int:
+    """Extracts version position from a Manuscript Tracking No.
+
+    example:
+
+    >>> extract_version_position('01-02-2017-SR-eLife-10002R2')
+    2
+    >>> extract_version_position('01-02-2017-SR-eLife-10002R1')
+    1
+    >>> extract_version_position('01-02-2017-SR-eLife-10002')
+    0
+
+    :param input_data: str
+    :return: int
+    """
+    position = 0
+    result = re.search(r'(R\d{1,2})', input_data)
+    if result:
+        position = int(result.group().replace('R', ''))
+
+    return position
+
+
+def convert_row(row: List[str], source_file: str, create_date: int) -> List[str]:
```

Comment:
I think it might be worth to check the headers. The SQL can easily be edited and someone might just add or insert another column. I'd rather it fail than silently populate the wrong column in that case. What do you think?

Reply:
What would you suggest is a reasonable check?

I agree that the source data structure could change (i.e. addition of a column) but this is just a purpose built script to handle converting that specific data.

That only check I would suggest could be added with ease would be to check the source header row is what we expect, otherwise raise an exception.

Reply:
Yes, similar. I would suggest to check the source header and pick out the header columns we need. Raise an exception if any of them are missing (or had been renamed). But be tolerant to additional columns that we don't care about and also the column order.

Does that seem reasonable?

Reply:
I have now implemented the suggested check. 👍